lib/src/default_index/revset_engine.rs

```diff
@@ -957,6 +957,21 @@ impl EvaluationContext<'_> {
                 let candidate_set = self.evaluate(candidates)?;
                 Ok(Box::new(self.take_latest_revset(&*candidate_set, *count)?))
             }
+            ResolvedExpression::GreatestCommonAncestors(expression) => {
+                let expression_set = self.evaluate(expression)?;
+                let mut expression_positions_iter = expression_set.positions().attach(index);
+                let Some(position) = expression_positions_iter.next() else {
+                    return Ok(Box::new(EagerRevset::empty()));
+                };
+                let mut positions = vec![position?];
+                for position in expression_positions_iter {
+                    positions = index
+                        .common_ancestors_pos(&positions, [position?].as_slice())
+                        .into_iter()
+                        .collect_vec();
+                }
+                Ok(Box::new(EagerRevset { positions }))
```

Comment:
Perhaps, the order is reversed because `common_ancestors_pos()` returns `BTreeSet`?

Reply:
Ah, because all revsets must return items in reverse index position?